ophyd/ophydobj.py

```diff
@@ -121,6 +124,14 @@ def __init__(self, *, name=None, attr_name='', parent=None, labels=None,
             name = self.name
         # Instantiate logger
         self.log = logging.getLogger(base_log + '.' + name)
+        self.__register_instance(self)
+
+    def __register_instance(self):
```

Comment:
make this a `@classmethod` to make sure we don't get different callbacks on different objects?